src/agent/src/message_queue_utils.cpp

```diff
@@ -48,3 +46,34 @@ void PushCommandsToQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue, const
         }
     }
 }
+
+std::optional<command_store::Command> GetCommandFromQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue)
+{
+    if (multiTypeQueue->isEmpty(MessageType::COMMAND))
+    {
+        return std::nullopt;
+    }
+
+    Message m = multiTypeQueue->getNext(MessageType::COMMAND);
+    nlohmann::json jsonData = m.data.at(0).at("data");
+
+    std::string id = jsonData["id"].dump();
+
+    // remove quotation marks from id
```

Comment:
Is this necessary?

Reply:
I found that using ```nlohmann::json::get<std::string>()``` instead of ```dump()``` returns the value without quotation marks. Yet for the parameters member we still need to use ```dump()``` probably because it's an array.